mypy/semanal.py

```diff
@@ -1696,51 +1694,53 @@ def build_namedtuple_typeinfo(self, name: str, items: List[str],
         tup = TupleType(types, self.named_type('__builtins__.tuple', types))
         class_def = ClassDef(name, Block([]))
         class_def.fullname = self.qualified_name(name)
-
         info = namedtuple_type_info(tup, symbols, class_def)
 
         vars = [Var(item, typ) for item, typ in zip(items, types)]
         this_type = self_type(info)
-        add_field = partial(self.add_namedtuple_field, symbols, info)
-        strtype = self.named_type('__builtins__.str')
+        strtype = self.named_type('__builtins__.str')  # type: Type
+
+        def add_field(var: Var, is_initialized_in_class: bool = False,
+                      is_property: bool = False) -> None:
+            var.info = info
+            var.is_initialized_in_class = is_initialized_in_class
+            var.is_property = is_property
+            symbols[var.name()] = SymbolTableNode(MDEF, var)
 
         for var in vars:
             add_field(var, is_property=True)
 
         tuple_of_strings = TupleType([strtype for _ in items],
                                      self.named_type('__builtins__.tuple', [AnyType()]))
-        add_field(Var('_fields', tuple_of_strings),
-                  is_initialized_in_class=True)
-        add_field(Var('_field_types', UnboundType('Dict', [strtype, AnyType()])),
-                  is_initialized_in_class=True)
-        add_field(Var('_source', strtype),
-                  is_initialized_in_class=True)
-
-        add_method = partial(self.add_namedtuple_method, symbols, info, this_type)
+        # TODO: refine type of values
+        dictype = self.external_instance('typing.Dict', [strtype, AnyType()])
+        add_field(Var('_fields', tuple_of_strings), is_initialized_in_class=True)
+        add_field(Var('_field_types', dictype), is_initialized_in_class=True)
+        add_field(Var('_source', strtype), is_initialized_in_class=True)
+
+        def add_method(funcname: str, ret: Type, args: List[Argument], name=None,
+                       is_classmethod=False) -> None:
+            self.add_namedtuple_method(symbols, info, this_type,
+                                       funcname, ret, args, name, is_classmethod)
+
         add_method('_replace', ret=this_type,
                    args=self.factory_args(vars, ARG_NAMED, initializer=EllipsisExpr()))
         add_method('__init__', ret=NoneTyp(),
                    args=self.factory_args(vars, ARG_POS), name=info.name())
-        # TODO: refine to OrderedDict[str, Union[types]]
-        add_method('_asdict', ret=UnboundType('collections.OrderedDict', is_ret_type=True),
-                   args=[])
-
-        # TODO: refine signature
-        union = UnboundType('Iterable', [UnionType(types)])
+        # actual signature should return OrderedDict[str, Union[types]]
+        dictype = self.external_instance('typing.Dict', [strtype, AnyType()])
```

Comment:
It does, but it means adding this fixture to every test case (in some, replacing `list`), regardless of actual calls to `_asdict()`, and getting strange errors when you forget - even if it's secondary to the test (since there are cases outside `check-namedtuple.test`).

What's wrong with `typing.Dict`? It should always be there.

And `OrderedDict` is appropriate in `typing` too; we might call it `_OrderedDict` until it makes its way there. Given [PEP 520: Preserving Class Attribute Definition Order](https://www.python.org/dev/peps/pep-0520/) and [PEP 468: Preserving the order of **kwargs in a function](https://www.python.org/dev/peps/pep-0468/), I'll expect this type to occur more frequently inside the checker in the future.